esp-hal/src/gpio/mod.rs

```diff
@@ -105,7 +105,10 @@ pub(crate) struct PinGuard {
 impl crate::private::Sealed for PinGuard {}
 
 impl PinGuard {
-    pub(crate) fn new(pin: AnyPin<'_>) -> Self {
+    // Safety:
```

Comment:
This isn't a memory safety concern, the function doesn't actually need to be unsafe.